content/guest-post/modern-login-signup-form-tailwindcss-react/index.md

```diff
@@ -0,0 +1,773 @@
+
+  const handleChange=(e)=>setSignupState({...signupState,[e.target.id]:e.target.value});
+
+  const handleSubmit=(e)=>{
+    e.preventDefault();
+    console.log(signupState)
+    createAccount()
+  }
+
+  //handle Signup API Integration here
+  const createAccount=()=>{
+
+  }
+
+    return(
+        <form className="mt-8 space-y-6" onSubmit={handleSubmit}>
+        <div className="">
+        {
+                fields.map(field=>
+                        <Input
+                            key={field.id}
+                            handleChange={handleChange}
+                            value={signupState[field.id]}
+                            labelText={field.labelText}
+                            labelFor={field.labelFor}
+                            id={field.id}
+                            name={field.name}
+                            type={field.type}
+                            isRequired={field.isRequired}
+                            placeholder={field.placeholder}
+                    />
+                
+                )
+            }
+          <FormAction handleSubmit={handleSubmit} text="Signup" />
+        </div>
+
+         
+
+      </form>
+    )
+}
+```
+
+Then, we'll render this component inside `/src/pages/Signup.js` alongisde the `<Header/>` components with some modified props for the signup page. 
+
+```jsx
+import Header from "../components/Header";
+import Signup from "../components/Signup";
+
+export default function SignupPage(){
+    return(
+        <>
+            <Header
+              heading="Signup to create an account"
+              paragraph="Already have an account? "
+              linkName="Login"
+              linkUrl="/"
+            />
+            <Signup/>
+        </>
+    )
+}
+```
+
+And that's it! That's all there is to building a great looking signup page that looks like this:
+
+
+
+![Signup page](./Signup%20page.png)
+
+Don't worry, if you got stuck somewhere, feel free to refer to the entire codebase for this tutorial here. 
+
+## Use LoginRadius Authentication APIs
+
+You can use [LoginRadius](https://www.loginradius.com/) to quickly setup an authentication backend for your frontend application. First, you'll need to setup an account and get your API key and secret from your account Dashboard. Then, you can simply use their [login endpoint](https://api.loginradius.com/identity/v2/auth/login) to authenticate users. 
+
+The endpoint takes an email and password as parameters and validates them against them against the correct credentials. If we were to integrate this enpoint in our previously build `<Login/>` component (`/scr/components/Login.js`) here's, how the `authenticateUser` method would look like:
+
+```javascript
+//Handle Login API Integration here
+    const authenticateUser = () =>{
+        
+       
+        const endpoint=`https://api.loginradius.com/identity/v2/auth/login?apikey=${apiKey}&apisecret=${apiSecret}`;
```

Comment:
@FuzzySid  No need to pass apisecret for authentication apis. Refer Document : https://www.loginradius.com/docs/api/v2/customer-identity-api/authentication/auth-login-by-email/